server/src/main/java/org/elasticsearch/common/logging/JULBridge.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.logging;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+import java.util.stream.Collectors;
+
+/**
+ * A Java Util Logging handler that writes log messages to log4j.
+ */
+class JULBridge extends Handler {
+
+    private static final Map<java.util.logging.Level, Level> levelMap = Map.of(
+        java.util.logging.Level.OFF,
+        Level.OFF,
+        java.util.logging.Level.SEVERE,
+        Level.ERROR,
+        java.util.logging.Level.WARNING,
+        Level.WARN,
+        java.util.logging.Level.INFO,
+        Level.INFO,
+        java.util.logging.Level.FINE,
+        Level.DEBUG,
+        java.util.logging.Level.FINEST,
+        Level.TRACE,
+        java.util.logging.Level.ALL,
+        Level.ALL
+    );
+
+    private static final TreeMap<Integer, Level> sortedLevelMap = new TreeMap<>(
```

Comment:
You could also write:
```java
 sortedLevelMap = levelMap.entrySet().stream().collect(Collectors.toMap(e -> e.getKey().intValue(), Map.Entry::getValue, (k1, k2) -> throw new UnsupportedOperationException(), TreeMap::new));
```

(I hate that theres no Collector method that allows to specify custom map factory, but uses default merge function. 🤮

Reply:
Thanks! I used the Maps method and changed to NavigableMap